system/lib/pthread/library_pthread.c

```diff
@@ -806,10 +656,7 @@ int emscripten_dispatch_to_thread_async_args(pthread_t target_thread,
     q->calleeDelete = 1;
 
     // Schedule the call to run later on this thread.
-    DispatchToThreadArgs* args = malloc(sizeof(DispatchToThreadArgs));
-    args->target_thread = target_thread;
-    args->q = q;
-    emscripten_set_timeout(dispatch_to_thread_helper, 0, args);
+    emscripten_set_timeout(_do_call, 0, args);
```

Comment:
Won't `args` (the va_list) be invalid once the timeout fires?  i.e. isn't this going to be a use-after-free?

Reply:
Ah yes, and `_do_call` expects an `em_queued_call*`, not a `va_list`. `args` should be `q` here.